### ISSUE TITLE
pre-alpha manifest merger skips manifests

### ISSUE THREAD

**Issue by katre** [bazelbuild/rules_android#77]
Discovered while testing https://github.com/bazelbuild/examples/pull/287.

When building examples/android/ndk with rules_android master branch, the build succeeds. The `ManifestMerger` action has the `--mergeeManfests` argument set to a list of manifest files, which includes `@maven//\:androidx_appcompat_appcompat,bazel-out/android-arm64-v8a-fastbuild-ST-0b94fd089017/bin/external/maven/androidx_appcompat_appcompat_resources_processed_manifest/AndroidManifest.xml`

When building with the pre-alpha branch, the `ManifestMerger` does not set the flag, and the merge action fails (presumably due to the lack of needed dependencies).

**Comment by katre** [comment]
Looks like the issue is [`RourceDependencies.fromRuleDeps`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/android/ResourceDependencies.java?q=symbol%3A%5Cbcom.google.devtools.build.lib.rules.android.ResourceDependencies.fromRuleDeps%5Cb%20case%3Ayes), which is looking for the [`AndroidResourceInfo.PROVIDER`](https://cs.opensource.google/bazel/bazel/+/refs/heads/master:src/main/java/com/google/devtools/build/lib/rules/android/AndroidResourcesInfo.java?q=symbol%3A%5Cbcom.google.devtools.build.lib.rules.android.AndroidResourcesInfo.PROVIDER%5Cb%20case%3Ayes) provider. This is present with the master branch, but not with the pre-alpha, which I assume is due to starlarkifying `android_library`.

**Comment by katre** [comment]
Yes, looks like the native `android_library` calls [`AndroidResources.from`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/android/AndroidLibrary.java;drc=01183bcaae4a77b6c7577aedda02cd5aa4ef0755;l=159), which collects the `AndroidResourceInfo` providers, but the version in pre-alpha does not.

**Comment by katre** [comment]
Okay, finally figured out the issue.

The pre-alpha version of resource collection (as implemented in [`rules/resources.bzl`](https://github.com/bazelbuild/rules_android/blob/pre-alpha/rules/resources.bzl#LL1282C76-L1282C76) only checks for the existence of `StarlarkAndroidResourcesInfo`, not the native `AndroidResourcesInfo`.

The Starlark version of `aar_import` creates this provider, as expected.

However, the code in examples/android/ndk depends on appcompat via [`maven_install`](https://github.com/bazelbuild/examples/blob/main/android/ndk/WORKSPACE#L16) from `rules_jvm_external`, which creates an `aar_import` target using the native rules, which uses the native provider.

Either the pre-alpha branch needs to check for both possible providers, or rules_jvm_export needs to use the same pre-alpha version of aar_import, or rules_jvm_export isn't compatible with rules_android pre-alpha.

**Comment by katre** [comment]
Looks like rules_jvm_external has [support for this, possibly](https://github.com/bazelbuild/rules_jvm_external/blob/master/docs/api.md#maven_install-use_starlark_android_rules).

**Comment by katre** [comment]
But even using the starlark `aar_import`, the target `@maven//:androidx_appcompat_appcompat` doesn't have the `StarlarkAndroidResourcesInfo` provider.

Using native `aar_import` to list provider names:
```
$ /tmp/blazes/bazel cquery --output=starlark --starlark:expr='providers(target).keys()' @maven//:androidx_appcompat_appcompat_resources
[
  "InstrumentedFilesInfo",
  "AndroidResourcesInfo",
  "AndroidManifestInfo",
  "AndroidAssetsInfo",
  "DataBindingV2Info",
  "ProguardSpecProvider",
  "AndroidNativeLibsInfo",
  "JavaInfo",
  "FileProvider",
  "FilesToRunProvider",
  "OutputGroupInfo",
]
```

Using Starlark `aar_import` to list provider names:
```
$ /tmp/blazes/bazel cquery --output=starlark --starlark:expr='providers(target).keys()' @maven//:androidx_appcompat_appcompat_resources
[
  "AndroidLibraryResourceClassJarProvider",
  "JavaInfo",
  "AndroidNativeLibsInfo",
  "ProguardSpecProvider",
  "AndroidIdeInfo",
  "FileProvider",
  "FilesToRunProvider",
  "OutputGroupInfo",
]
```

**Comment by ted-xie** [comment]
This should be fixed if/when we merge https://github.com/bazelbuild/rules_android/pull/52/files#diff-e2839f4bd7500de02db70226d1fb363bc1bcd62e2ac7046594dafc3e7040fa09.

**Comment by ThomasCJY** [comment]
Is this issue fixed in https://github.com/bazelbuild/rules_android/issues/224?
